index.js

```diff
@@ -1,159 +1,201 @@
+if (!DATABASE_HOST) {
+  throw new Error("DATABASE_HOST not set!");
 }
-if(!TEAM) {
-  throw new Error('TEAM not set!');
+if (!DATABASE_USERNAME) {
+  throw new Error("DATABASE_USERNAME not set!");
 }
-if(!DATABASE_HOST) {
-  throw new Error('DATABASE_HOST not set!');
+if (!DATABASE_PASSWORD) {
+  throw new Error("DATABASE_PASSWORD not set!");
 }
-if(!DATABASE_USERNAME) {
-  throw new Error('DATABASE_USERNAME not set!');
+if (!DATABASE_NAME) {
+  throw new Error("DATABASE_NAME not set!");
 }
-if(!DATABASE_PASSWORD) {
-  throw new Error('DATABASE_PASSWORD not set!');
+if (!KC_HOST) {
+  throw new Error("KC_HOST not set!");
 }
-if(!DATABASE_NAME) {
-  throw new Error('DATABASE_NAME not set!');
+if (!KC_REALM) {
+  throw new Error("KC_REALM not set!");
 }
-if(!GH_CLIENT_ID) {
-  throw new Error('GH_CLIENT_ID not set!');
+if (!KC_CLIENT_ID) {
+  throw new Error("KC_CLIENT_ID not set!");
 }
-if(!GH_CLIENT_SECRET) {
-  throw new Error('GH_CLIENT_SECRET not set!');
+if (!KC_CLIENT_SECRET) {
+  throw new Error("KC_CLIENT_SECRET not set!");
 }
-if(!GH_CALLBACK_URL) {
-  throw new Error('GH_CALLBACK_URL not set!');
+if (!ADMIN_ACCESS_TOKEN) {
+  throw new Error("ADMIN_ACCESS_TOKEN not set!");
 }
-if(!ADMIN_ACCESS_TOKEN) {
-  throw new Error('ADMIN_ACCESS_TOKEN not set!');
+if (!CLIENT_ACCESS_TOKEN) {
+  throw new Error("CLIENT_ACCESS_TOKEN not set!");
 }
-if(!CLIENT_ACCESS_TOKEN) {
-  throw new Error('CLIENT_ACCESS_TOKEN not set!');
+if (!SESSION_SECRET) {
+  throw new Error("SESSION_SECRET not set!");
 }
-if(!SESSION_SECRET) {
-  throw new Error('SESSION_SECRET not set!');
-}
-
 
-function adminAuth(app, config, services) {
+function enableKeycloakOauth(app, config, services) {
   const { baseUriPath } = config.server;
   const { userService, accessService } = services;
-  
+
   passport.use(
-    new githubStrategy(
+    "keycloak",
+    new KeycloakStrategy(
       {
-        clientID: GH_CLIENT_ID,
-        clientSecret: GH_CLIENT_SECRET,
-        callbackURL: GH_CALLBACK_URL,
+        host: KC_HOST,
+        realm: KC_REALM,
+        clientID: KC_CLIENT_ID,
+        clientSecret: KC_CLIENT_SECRET,
+        callbackURL: `${baseUriPath}/api/auth/callback`,
+        authorizationURL: `${KC_HOST}/realms/${KC_REALM}/protocol/openid-connect/auth`,
+        tokenURL: `${KC_HOST}/realms/${KC_REALM}/protocol/openid-connect/token`,
+        userInfoURL: `${KC_HOST}/realms/${KC_REALM}/protocol/openid-connect/userinfo`,
       },
-      async (accessToken, refreshToken, profile, cb) => {
-  
-        var role;
-  
-        for (var index in profile.teams) {
-          if (READ_ONLY_TEAM && profile.teams[index].name === READ_ONLY_TEAM) {
-            role = "Viewer";
-          }
-          if (profile.teams[index].name === TEAM) {
-            role = "Editor";
-            break;
-          }
-        }
-  
-        if (!role) {
-          cb(null, false);
+
+      async (accessToken, refreshToken, profile, done) => {
+        console.log("profile", profile);
```

Comment:
What's in `profile`, is that safe to log like PII data?

Reply:
There are just non-confidential data (email, name, roles, ...), debugging user login would be easier with that.

Reply:
Is that OK to log email as it's PII, can we use some id to debug?

Reply:
TBH, I don't want to die on that hill. What do you think about cef8bcc?

Reply:
yes, that should be good enough to debug, thanks!